src/GraphQL/Types/GraphTypesLookup.cs

```diff
@@ -239,7 +245,7 @@ public void AddType(IGraphType type, TypeCollectionContext context)
             {
                 foreach (var field in complexType.Fields)
                 {
-                    HandleField(type.GetType(), field, context);
+                    HandleField(complexType, field, context, false);
```

Comment:
I noticed that control gets here for the nested fields of introspection, not just `__XXX`. `introspectionType` may confuse a bit. The converter should not affect not only `__XXX` fields, but also all nested fields and their arguments - the entire introspection graph. If I understand correctly, now the converter **works** for all the nested introspection fields, but since it is equal to `CamelCaseNameConverter.Instance`, it does not produce any effect. Right?

Reply:
I think I understand what you're saying.  Yes the converter works and is operational for all nested introspection fields.  The property you referred to, `introspectionType` is like an override, specifically for the three root introspection query fields -- not fields of an introspection type.  The parameter is necessary because at the specific time that HandleField is called for these "query fields", the `NameFieldConverter` property has already been set to the schema's selected name converter.  When this boolean property is true, HandleField will ignore the `NameFieldConverter` altogether, leaving the field names alone.  This only applies to the three root introspection query fields -- `__schema`, `__type`, and `__typename`.  It does not apply to any of the introspection types or their fields.

I've renamed the field to `applyNameConverter`, inversed the logic, and added a boatload of comments to help clarify this for others in the future.

By the way, this design isn't mine; this is the way it was when I found it.  I just changed it from passing a `Type` and then comparing `type == typeof(SchemaIntrospection)` to a simple boolean, since the parameter served no other purpose.

Reply:
OK.